quic/s2n-quic-core/src/recovery/bandwidth/estimator.rs

```diff
@@ -26,45 +30,71 @@ pub struct PacketInfo {
     pub is_app_limited: bool,
 }
 
-const MICRO_BITS_PER_BYTE: u64 = 8 * 1000000;
-
-#[derive(Copy, Clone, Debug, Default, Eq, Ord, PartialOrd, PartialEq)]
+/// Represents a rate at which data is transferred
+///
+/// While bandwidth is typically thought of as an amount of data over a fixed
+/// amount of time (bytes per second, for example), in this case we internally
+/// represent bandwidth as the inverse: an amount of time to send a fixed amount
+/// of data (nanoseconds per byte, in this case). This allows for some of the  
+/// math operations needed on `Bandwidth` to avoid division, while reducing the
+/// likelihood of panicking due to overflow.
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
 pub struct Bandwidth {
-    bits_per_second: u64,
+    nanos_per_byte: u64,
 }
 
 impl Bandwidth {
-    pub const ZERO: Bandwidth = Bandwidth { bits_per_second: 0 };
-
-    pub const MAX: Bandwidth = Bandwidth {
-        bits_per_second: u64::MAX,
+    pub const ZERO: Bandwidth = Bandwidth {
+        nanos_per_byte: u64::MAX,
     };
 
+    pub const INFINITY: Bandwidth = Bandwidth { nanos_per_byte: 0 };
+
     /// Constructs a new `Bandwidth` with the given bytes per interval
     pub const fn new(bytes: u64, interval: Duration) -> Self {
-        if interval.is_zero() {
+        if interval.is_zero() || bytes == 0 {
             Bandwidth::ZERO
```

Comment:
doesnt interval == 0 mean Infinity? or is that not a realistic value we want to allow for actual usecase?

Reply:
hmm, yeah it might be better to just make that a debug assertion, interval == 0 isn't really valid

Reply:
actually, looks like I can't add a debug assertion to a const function until we increase our minimum rust version:

```rust
error[E0658]: panicking in constant functions is unstable
  --> quic/s2n-quic-core/src/recovery/bandwidth/estimator.rs:57:9
   |
57 |         debug_assert!(!interval.is_zero());
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #51999 <https://github.com/rust-lang/rust/issues/51999> for more information
```

Reply:
Its allowed in Rust 1.57.0:  https://blog.rust-lang.org/2021/12/02/Rust-1.57.0.html#panic-in-const-contexts